src/main/java/com/github/manolo8/darkbot/gui/MainGui.java

```diff
@@ -33,8 +37,24 @@ public MainGui(Main main) throws HeadlessException {
         ToolTipManager.sharedInstance().setInitialDelay(350);
 
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        setSize(640, 480);
-        setLocationRelativeTo(null);
+
+        Config.BotSettings botSettings = main.config.BOT_SETTINGS;
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            if (botSettings.SAVE_MAIN_GUI_POS_AND_SIZE)
+                main.configManager.saveConfig();
+        }));
+
+        Config.BotSettings.Window thisWindow = botSettings.MAIN_GUI_WINDOW;
+        if (!botSettings.SAVE_MAIN_GUI_POS_AND_SIZE ||
+                (thisWindow.x == Integer.MIN_VALUE && thisWindow.y == Integer.MIN_VALUE) ||
```

Comment:
```suggestion
```
this check shouldn't be needed at all, it will be outside screen if that happens